src/js/containers/search/visualizations/time/TimeVisualizationSectionContainer.jsx

```diff
@@ -34,44 +36,104 @@ export class TimeVisualizationSectionContainer extends React.Component {
+        this.setFilterStates();
     }
 
     componentDidUpdate(prevProps) {
         if (!_.isEqual(prevProps.reduxFilters, this.props.reduxFilters)) {
-            this.fetchData();
+            this.setFilterStates();
         }
     }
 
-    fetchData() {
-        // build a new search operation from the Redux state, but create a transaction-based search
-        // operation instead of an award-based one
-        const operation = new SearchTransactionOperation();
-        operation.fromState(this.props.reduxFilters);
+    setFilterStates() {
+        this.setState({
+            budgetFiltersSelected: SearchHelper.budgetFiltersSelected(this.props.reduxFilters),
+            awardFiltersSelected: SearchHelper.awardFiltersSelected(this.props.reduxFilters)
+        }, () => {
+            this.fetchData();
+        });
+    }
 
+    createTransactionsParams(operation) {
         const searchParams = operation.toParams();
 
         // generate the API parameters
         const apiParams = {
             field: 'federal_action_obligation',
             group: 'action_date__fy',
-            order: ['item'],
+            order: ['aggregate'],
             aggregate: 'sum',
             filters: searchParams
         };
 
+        return apiParams;
+    }
+
+    createTasParams() {
+        const operation = new SearchAccountOperation('appropriations');
+        operation.fromState(this.props.reduxFilters);
+
+        const searchParams = operation.toParams();
+
+        // generate the API parameters
+        const apiParams = {
+            field: 'obligations_incurred_total_by_tas_cpe',
+            group: 'submission__reporting_fiscal_year',
+            order: ['aggregate'],
+            aggregate: 'sum',
+            filters: searchParams
+        };
+
+        return apiParams;
+    }
+
+    fetchData() {
         this.setState({
             loading: true
         });
-        const search = SearchHelper.performTransactionsTotalSearch(apiParams);
-        search.promise
+
+        if (this.apiRequest) {
+            this.apiRequest.cancel();
+        }
+
+        let apiParams = '';
+
+        // Create Search Operation and API request
+        if (this.state.awardFiltersSelected) {
+            let operation = null;
+
+            if (this.state.budgetFiltersSelected) {
+                operation = new SearchTransactionFileCOperation();
+            }
+            else {
+                operation = new SearchTransactionOperation();
+            }
+
+            operation.fromState(this.props.reduxFilters);
+            apiParams = this.createTransactionsParams(operation);
+            this.apiRequest = SearchHelper.performTransactionsTotalSearch(apiParams);
+        }
+        else {
+            apiParams = this.createTasParams();
+            this.apiRequest = SearchHelper.performBalancesSearch(apiParams);
+        }
+
+        this.apiRequest.promise
             .then((res) => {
                 this.parseData(res.data);
```

Comment:
Does `this.parseData` need to be refactored to account for the different API calls this promise could reference?